src/modules/secrets/secrets.service.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+  NotFoundException,
+} from '@nestjs/common';
+import { SecretsDAL } from './secrets.dal';
+import { AccountDAL } from '../account/account.dal';
+import { CreateSecretsDto } from './dtos/create-secrets.dto';
+import { UpdateSecretsDto } from './dtos/update-Secrets.dto';
+import { encrypt, decrypt } from 'src/common/utils/encrypt';
+
+@Injectable()
+export class SecretsService {
+  constructor(
+    private readonly secretsDAL: SecretsDAL,
+    private readonly accountsDAL: AccountDAL,
+  ) {}
+
+  async createSecret(createSecretDto: CreateSecretsDto, accountId: number) {
+    try {
+      const account = await this.accountsDAL.findAccountById(accountId);
+      if (!account) {
+        throw new NotFoundException('Account not found');
+      }
+      const encryptedValue = encrypt(createSecretDto.value, account.password);
+
+      const createdSecret = await this.secretsDAL.createSecret(
+        { ...createSecretDto, value: encryptedValue },
+        accountId,
+      );
+      if (!createdSecret) {
+        throw new InternalServerErrorException('Failed to create secret.');
+      }
+      return createdSecret;
+    } catch {
+      throw new BadRequestException('Could not create secret');
+    }
+  }
+
+  async findAllSecretsByAccount(accountId: number) {
+    try {
+      const secrets = await this.secretsDAL.findAllSecrets(accountId);
+      if (secrets.length === 0) {
+        throw new NotFoundException('No secrets found for this account.');
```

Comment:
@danielaKokoneshi309 I wouldn't throw en error in this case. This is different from asking for a specific key and not finding it. If I want all the keys for an account and I don't have any, I would expect an empty array instead of an error